src/colors/RGBColor.ts

```diff
@@ -0,0 +1,146 @@
+import { vec3 } from 'gl-matrix';
+import { Color } from './Color';
+
+/**
+ * An RGB representation of a color in a WebGL context.
+ *
+ * The acronym stands for:
+ *   R: `R`ed
+ *   G: `G`reen
+ *   B: `B`lue
+ *
+ * @class RGBColor
+ */
+export class RGBColor implements Color {
+    private red: number;
+    private green: number;
+    private blue: number;
+
+    /**
+     * Instantiate a new RGBColor object.
+     *
+     * @class RGBColor
+     * @constructor
+     * @param {number} red Red numeric color value.
+     * @param {number} green Green numeric color value.
+     * @param {number} blue Blue numeric color value.
+     */
+    private constructor(red: number, green: number, blue: number) {
+        if (red < 0 || red > 255) {
+            throw new RangeError(`red's value (${red}) must be in range [0, 255]`);
+        } else if (green < 0 || green > 255) {
+            throw new RangeError(`green's value (${green}) must be in range [0, 255]`);
+        } else if (blue < 0 || blue > 255) {
+            throw new RangeError(`blue's value (${blue}) must be in range [0, 255]`);
+        }
+
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
+    }
+
+    /**
+     * Instantiates a new RGBColor object from a hex string.
+     *
+     * @class RGBColor
+     * @method fromHex
+     * @param {string} value A hex string of the form `FF0000` or `#FF0000`.
+     * @returns {RGBColor}
+     */
+    public static fromHex(value: string): RGBColor {
+        // Ensure the user passed in a valid hexadecimal string
+        if (
+            value.length > 7 ||
+            value.length < 6 ||
+            (value.length === 7 && value.lastIndexOf('#', 0) !== 0)
+        ) {
+            throw new Error('Please pass in a valid hexadecimal string (i.e., FF33AA or #FF33AA)');
```

Comment:
I think you can just throw this error in the `if (match === null)` check now, since the regex asserts all of this already

Reply:
ahh, gotcha - sounds good 👍